orchestra/static/orchestra/project/js/controllers.js

```diff
@@ -0,0 +1,870 @@
+            vis.prepareData();
+            dataService.data(vis.data);
+            vis.draw();
+          });
+      }
+
+      vis.prepareData = function() {
+        _.each(vis.data.tasks, function(task) {
+          _.each(task.assignments, function(assignment) {
+            assignment.task = task;
+            _.each(assignment.iterations, function(iteration, i) {
+              iteration.assignment = assignment;
+            })
+          })
+        })
+
+        var timeSortedTasks = _.toArray(vis.data.tasks).sort(function(a, b) {
+                                return d3.ascending(a.start_datetime.toDate(), b.start_datetime.toDate());
+                             });
+        vis.timeSortedSlugs = _.map(timeSortedTasks, function(task) {return task.step_slug})
+      }
+
+      vis.setupAxis = function() {
+        var firstTask = vis.data.tasks[_.first(vis.timeSortedSlugs)];
+        minDatetime = firstTask.start_datetime.toDate();
+
+        var taskEndTimes = _.map(vis.timeSortedSlugs,
+                                 function(slug) {return dataService.taskEnd(dataService.taskFromKey(slug))})
+        var maxDatetime = _.max(taskEndTimes,
+                                function(time) {return time.toDate()}).toDate()
+
+        var hourInMilliseconds = 60 * 60 * 1000;
+        var numHours = Math.ceil((maxDatetime - minDatetime) / hourInMilliseconds);
+        maxDatetime = new Date(minDatetime.getTime() + numHours * hourInMilliseconds);
+
+        hourTicks = _.map(d3.range(0, numHours + 1), function(hourIndex) {
+          return new Date(minDatetime.getTime() + hourIndex * hourInMilliseconds);
+        })
+
+        vis.timeScale = d3.time.scale()
+          .domain([minDatetime, maxDatetime])
+          .range([0, vis.scaleWidth]);
+
+        var xAxis = d3.svg.axis()
+          .scale(vis.timeScale)
+          .tickSize(10)
+
+        var tickSpread = 5;
+        var xLabelText;
+        if (vis.relativeTime) {
+          xAxis.tickValues(hourTicks)
+            .tickFormat(function(d, i) {
+              return i % tickSpread == 0 ? (d - minDatetime) / hourInMilliseconds : '';
+            });
+          xLabelText = 'Time (hours)'
+        }
+        else {
+          xAxis.ticks(d3.time.hour, 1);
+          var defaultFormat = vis.timeScale.tickFormat();
+          xAxis.tickFormat(function(d, i) {
+              return i % tickSpread == 0 ? defaultFormat(d) : '';
+            });
+          xLabelText = 'Time (local)'
+        }
+
+        vis.parentContainer.select('.x.label')
+          .text(xLabelText)
+          .style('right', vis.utils.getSvgWidth() + 5 + 'px');
+
+        vis.parentContainer.select('.x.axis').transition().call(xAxis);
+        d3.select('.x.axis')
+          .attr({
+            'width': vis.utils.getSvgWidth()
+          })
+      }
+
+      function tasksVis() {
+        var tasksVis = this;
+
+        tasksVis.draw = function() {
+          tasksVis.drawMeta();
```

Comment:
i see a few of these `drawMeta` around, and don't know what they do.  can you change the name to something more informative or add some comments?